test/test_helper.rb

```diff
@@ -0,0 +1,39 @@
+if ENV['RAILS_ENV'] ||= 'test'
+  require 'simplecov'
+  SimpleCov.start 'rails'
+  puts "required simplecov"
+  require_relative '../config/environment'
+  require 'rails/test_help'
+  require "minitest/rails"
+  require "minitest/reporters" # for Colorized output
+end
+Minitest::Reporters.use!(
+    Minitest::Reporters::SpecReporter.new,
+    ENV,
+    Minitest.backtrace_filter
+)
+
+class ActiveSupport::TestCase
+  # Run tests in parallel with specified workers
+  # parallelize(workers: :number_of_processors) # causes out of order output.
+
+  # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
+  fixtures :all
+
+  # Add more helper methods to be used by all tests here...
+
+  def perform_login(user = nil)
+    user ||= User.first
+
+    login_data = {
+        user: {
+            username: user.name,
+        },
+    }
```

Comment:
Notice that the form you use to log in doesn't nest `username` under a `user` hash.  Also in your controller you are using "name" as the key and not "username".

So this should be:

```suggestion
    login_data = {
          name: user.name,
    }
```